pkg/statistics/handle/autoanalyze/priorityqueue/queue_v2.go

```diff
@@ -203,26 +460,90 @@ func (pq *AnalysisPriorityQueueV2) RefreshLastAnalysisDuration() {
 	}
 }
 
+// GetRunningJobs returns the running jobs.
+// Note: This function is thread-safe.
+func (pq *AnalysisPriorityQueueV2) GetRunningJobs() map[int64]struct{} {
+	pq.mu.RLock()
+	defer pq.mu.RUnlock()
+
+	runningJobs := make(map[int64]struct{}, len(pq.runningJobs))
+	for id := range pq.runningJobs {
+		runningJobs[id] = struct{}{}
+	}
+	return runningJobs
+}
+
+// AddRunningJob adds a running job.
+// Note: This function is thread-safe.
+func (pq *AnalysisPriorityQueueV2) AddRunningJob(tableID int64) error {
+	pq.mu.Lock()
+	defer pq.mu.Unlock()
+	if !pq.initialized {
+		return errors.New(notInitializedErrMsg)
+	}
+
+	pq.runningJobs[tableID] = struct{}{}
+	return nil
+}
+
+// RemoveRunningJob removes a running job.
+// Note: This function is thread-safe.
+func (pq *AnalysisPriorityQueueV2) RemoveRunningJob(tableID int64) error {
```

Comment:
Is this function only used by test cases?I have a question that is there any place would shrink this PQ or this PQ just grow forever?

Reply:
Do we need to set a maximum size for this PQ? I'm worrying the performance and memory usage if we put to many objects into this PQ.

Reply:
> Is this function only used by test cases?

The analysis worker will use this after it finishes the analysis.

> I have a question that is there any place that would shrink this PQ or this PQ just grow forever?

If the analysis is too slow, it will grow slowly. But it depends on the user's workload.

> Do we need to set a maximum size for this PQ? I'm worrying the performance and memory usage if we put to many objects into this PQ.

No, I don't think so. Based on my testing, the memory consumption for 1 million tables is about 200 MiB. So, I believe it's acceptable to keep all of them. Typically, one cluster shouldn't have all 1 million tables needing analysis at the same time.

Reply:
`Based on my testing, the memory consumption for 1 million tables is about 200 MiB. So, I believe it's acceptable to keep all of them. Typically, one cluster shouldn't have all 1 million tables needing analysis at the same time.`
Got it, could you add your test result as comments into it? I think this can help us eliminate these concerns and give us some basic baseline about its performance.

Reply:
Added.